docs(clayui.com): Update documentation for menubar (vertical navigation)

ambrinchaudhary opened the pull request:
Fixes https://github.com/liferay/clay/issues/3999

Updated documentation for the Menubar (Vertical Navigation)

ambrinchaudhary commented:
Hey @matuzalemsteles @pat270, 

I'm not sure what is the proccess to document in Clay... but I found there was an existing documentation for this and I updated it following what is present in https://github.com/liferay/clay/blob/master/packages/clay-css/src/content/menu_bar_vertical.html, because that is what I saw in `typography` section.... 

Let me know if I should change it, document it in another way, if you miss something... 

Thanks!

matuzalemsteles commented:
Hey @pat270 you could take a look at this if that’s what you’re thinking.

@ambrinchaudhary the documents in `clay-css` are Patrick's smoke test site, which usually keeps them up to date but they are more up-to-date than on clayui.com, so don't worry about updating them.

pat270 commented:
@ambrinchaudhary I didn't realize we already had existing docs on this. We went through a couple iterations of clayui docs and I forgot we had CSS Framework > Content. We could move this to `content/docs/components/markup-menubar.md`. It should show up in the Components panel.

The one thing we should change back is adding back `#css-` prefix to id and href for table of contents links and headers. This will prevent duplicate id's if we turn this into a Clay React component.

pat270 commented on clayui.com/content/docs/css/content/menubar.md:
```diff
@@ -9,19 +9,16 @@ order: 5
-    -   [Menubar Transparent](#css-menubar-vertical-expand-md-transparent)
-    -   [Menubar Decorated](#css-menubar-vertical-expand-md-decorated)
--   [Menubar Vertical Expand Lg](#css-menubar-vertical-expand-lg)
+- [Menubar Vertical Expand Md](#menubar-vertical-expand-md)
```
Should add back `#css-` to these links

pat270 commented on clayui.com/content/docs/css/content/menubar.md:
```diff
@@ -9,19 +9,16 @@ order: 5
 </div>
 
-## Menubar Vertical Expand Md(#css-menubar-vertical-expand-md)
+## Menubar Vertical Expand Md
```
Should add back `(#css-menubar-vertical-expand-md)`

ambrinchaudhary commented:
Hey @pat270, so I added back `#css-` prefix and moved the doc to the component section. Please have a look and let me know if something needs to be changed. Thank you!

matuzalemsteles commented on clayui.com/content/docs/components/markup-menubar.md:
```diff
@@ -9,19 +9,15 @@ order: 5
-    -   [Menubar Transparent](#css-menubar-vertical-expand-md-transparent)
-    -   [Menubar Decorated](#css-menubar-vertical-expand-md-decorated)
--   [Menubar Vertical Expand Lg](#css-menubar-vertical-expand-lg)
+-   [Menubar Vertical Expand Md(#css-menubar-vertical-expand-md)](#menubar-vertical-expand-mdcss-menubar-vertical-expand-md) - [Menubar Vertical Expand Md with Buttons](#menubar-vertical-expand-md-with-buttons)
+-   [Menubar Vertical Expand Lg(#css-menubar-vertical-expand-lg)](#menubar-vertical-expand-lgcss-menubar-vertical-expand-lg)
```
Probably a format error 🙂

ambrinchaudhary replied:
Well... SF did this change... otherwise it was complaining. It also looks strange to me, but... 🤷‍♀️

matuzalemsteles replied:
haha no problem, sometimes SF formats wrong when it has the wrong tab. I can send a commit correcting this without any problems because otherwise, the ToC will get a bug.

ambrinchaudhary replied:
Sure, go ahead!

matuzalemsteles commented:
Hey @pat270 I'm going to merge it, to include in today's release, if you think you have any changes to be made PR, we can create a new PR to fix it.
